packages/x-charts/src/models/slots/chartsIconSlots.ts

```diff
@@ -0,0 +1,15 @@
+import * as React from 'react';
+import { IconProps } from './chartsBaseSlotProps';
+
+export interface ChartsIconSlots {
+  /**
+   * Icon displayed on the toolbar's zoom in button.
+   * @default ChartsZoomInIcon
+   */
+  zoomInIcon: React.ComponentType<IconProps>;
+  /**
+   * Icon displayed on the toolbar's zoom out button.
+   * @default ChartsZoomOutIcon
+   */
+  zoomOutIcon: React.ComponentType<IconProps>;
+}
```

Comment:
Should this be in the pro package?

Or something like 

```ts
export interface ChartsIconSlotsExtension {}
export interface ChartsIconSlots extends ChartsIconSlotsExtension { }
```

To allow doing 

```ts
declare module '@mui/x-charts' {
  interface ChartsIconSlotsExtension {
	  /**
	   * Icon displayed on the toolbar's zoom in button.
	   * @default ChartsZoomInIcon
	   */
	  zoomInIcon: React.ComponentType<IconProps>;
	  /**
	   * Icon displayed on the toolbar's zoom out button.
	   * @default ChartsZoomOutIcon
	   */
	  zoomOutIcon: React.ComponentType<IconProps>;
  }
}
```

A bit similar to https://github.com/mui/mui-x/issues/14063

Reply:
Done 👍

Reply:
I did it, but I'm now reconsidering. We are still exporting the pro icons from the community package, so there isn't a big benefit in extending the types, is there? 

Another option would be for the pro package to export a `ChartsSlotsPro` type which would extend from `ChartsSlots`. That way we could move the pro-only components to that package instead of increasing the bundle size of the community version. 

If you think that's a good idea, I could do it in a follow-up PR, what do you think?

Reply:
Moved the pro slots to the pro package